lib/qless/middleware/retry_exceptions.rb

```diff
@@ -48,19 +48,16 @@ def on_retry_callback
         @on_retry_callback ||= DEFAULT_ON_RETRY_CALLBACK
       end
 
-      def exponential(base, options = {})
+      # If`factor` is omitted it is set to `delay_s` to reproduce legacy
+      # behavior.
+      def exponential(delay_seconds, options={})
+        factor = options.fetch(:factor, delay_seconds)
         fuzz_factor = options.fetch(:fuzz_factor, 0)
 
-        lambda do |num, _error|
-          unfuzzed = base**num
-
-          fuzz = 0
-          unless fuzz_factor.zero?
-            max_fuzz = unfuzzed * fuzz_factor
-            fuzz = rand(max_fuzz) * [1, -1].sample
-          end
-
-          unfuzzed + fuzz
+        lambda do |retry_no, error|
+          unfuzzed = delay_seconds * factor**(retry_no - 1)
+          return unfuzzed if fuzz_factor.zero?
+          unfuzzed * (1 + fuzz_factor * (2 * rand  - 1))
```

Comment:
The first two points make sense.

I disagree on it being more explicit...I find this much, much more difficult to understand than what was there before.  IMO, what was there was much easier to understand.  I've had to stare at this formula for a long time to understand it.  That's just subjective, though; my brain understood the ealier logic very easily but has a harder time understanding what all the constants in your version represent.  I get it now, though, and it's clearly better so let's go with it.